Userland/Utilities/sort.cpp

```diff
@@ -1,45 +1,126 @@
 /*
  * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, Peter Elliott <pelliott@serenityos.org>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #include <AK/QuickSort.h>
 #include <AK/String.h>
 #include <AK/Vector.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
 #include <LibMain/Main.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
-#include <unistd.h>
+#include <ctype.h>
+
+struct Line {
+    StringView key;
+    long int numeric_key;
+    String line;
+    bool numeric;
+
+    bool operator<(Line const& other) const
+    {
+        if (numeric)
+            return numeric_key < other.numeric_key;
+
+        return key < other.key;
+    }
+
+    bool operator==(Line const& other) const
+    {
+        if (numeric)
+            return numeric_key == other.numeric_key;
+
+        return key == other.key;
+    }
+
+private:
+};
+
+template<>
+struct AK::Traits<Line> : public GenericTraits<Line> {
+    static unsigned hash(Line l)
+    {
+        if (l.numeric)
+            return l.numeric_key;
+
+        return l.key.hash();
+    }
+};
+
+struct Options {
+    size_t key_field { 0 };
+    bool unique { false };
+    bool numeric { false };
+    StringView seperator { "\0", 1 };
```

Comment:
This is a mistake that I often make as well, but it turns out that the correct spelling is "separator".

Reply:
Also, couldn't one just initialize from `"\0"sv` here? The constructor that constructs from a string literal shouldn't be using `strlen`, right?

Reply:
both done